pkg/controller/logstorage/users/users_controller.go

```diff
@@ -357,15 +358,51 @@ func (r *UsersCleanupController) Reconcile(ctx context.Context, request reconcil
 		request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
 	reqLogger.Info("Reconciling LogStorage - Cleanup")
 
-	// Wait for Elasticsearch to be installed and available.
-	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
-	if err != nil {
+	// We skip requests without a namespace specified in multi-tenant setups.
+	if r.multiTenant && request.Namespace == "" {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if this is a tenant-scoped request.
+	_, _, err := utils.GetTenant(ctx, r.multiTenant, r.client, request.Namespace)
+	if errors.IsNotFound(err) {
+		reqLogger.Info("No Tenant in this Namespace, skip")
+		return reconcile.Result{}, nil
+	} else if err != nil {
 		return reconcile.Result{}, err
 	}
-	if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+
+	// Get LogStorage resource.
+	logStorage := &operatorv1.LogStorage{}
+	err = r.client.Get(ctx, utils.DefaultTSEEInstanceKey, logStorage)
+	if err != nil {
+		// Not finding the LogStorage CR is not an error, as a Managed cluster will not have this CR available but
+		// there are still "LogStorage" related items that need to be set up
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		reqLogger.Info("Logstorage is not found")
+		return reconcile.Result{}, nil
+	}
+
+	// Wait for the initializing controller to indicate that the LogStorage object is actionable.
+	if logStorage.Status.State != operatorv1.TigeraStatusReady {
+		reqLogger.Info("Logstorage is not marked as ready")
 		return reconcile.Result{}, nil
 	}
 
+	if !r.elasticExternal {
```

Comment:
From the description of the PR, it sounds like this is the only change necessary to fix the bug?

Reply:
Yup. I added the rest of the code to have parity with the user creation controller, but I am not sure if that is really needed since we read all tenant anyway for deletion. I am happy to only have this put in at the moment.